src/module/service.cpp

```diff
@@ -198,6 +199,10 @@ ModuleService_::ModuleService_(std::string addr) {
     module = std::make_shared<Module>(addr);
 }
 
+void ModuleService_::signal_handler(int) {
+    exit(0);
```

Comment:
That's a good point; I'll call the destructor directly instead of exiting. For others' [reference](https://en.cppreference.com/w/cpp/utility/program/exit) vis-a-vis `exit` and destructors.

Reply:
Calling the destructor directly does not actually seem feasible, as signal handling functions must be `static` and therefore cannot be associated with a particular instance of a module (no access to `this` and no access to the destructor). If we want to handle signals from within the library we'll probably need a more complex solution here with a thread that checks whether `SIGTERM` or `SIGINT` has been called periodically.